DeusEx/Classes/DXRandoCrowdControlLink.uc

```diff
@@ -0,0 +1,1046 @@
+
+    //Clean up move speed, since it seems to follow through levels
+    dxr.Player.Default.GroundSpeed = DefaultGroundSpeed;
+    
+    //Clean up LAM Throwers
+    UndoLamThrowers();
+    
+    //Clean up ice physics
+    SetIcePhysics(False);
+    
+    //Clean up third-person view
+    dxr.Player.bBehindView=False;
+    
+    //Clean up damage multiplier
+    dxr.Player.MPDamageMult = 1.0;
+    
+
+}
+
+function StopMatrixMode(optional bool silent) {
+    if (dxr.Player.Sprite!=None) {
+        dxr.Player.Matrix();
+    }
+
+    if (!silent) {
+        PlayerMessage("Your powers fade...");
+    }
+
+}
+
+function Timer() {
+    
+    ticker++;
+    
+    if (IsConnected()) {
+        ManualReceiveBinary();
+    }
+
+    if (ticker%10 != 0) {
+        return;
+    }
+    //Everything below here runs once a second
+
+    if (!IsConnected()) {
+        reconnectTimer-=1;
+        if (reconnectTimer <= 0){
+            Resolve(crowd_control_addr);
+        }
+    } 
+
+    //Matrix Mode Timer
+    if (matrixModeTimer>0) {
+        matrixModeTimer-=1;
+        if (matrixModeTimer == 0) {
+            StopMatrixMode();
+        }
+    }
+
+    //EMP Field timer
+    if (empTimer>0) {
+        empTimer -= 1;
+        if (empTimer <=0) {
+            dxr.Player.bWarrenEMPField = false;
+            PlayerMessage("EMP Field has disappeared...");
+        }
+    }
+
+    if (jumpTimer>0) {
+        jumpTimer -= 1;
+        dxr.Player.JumpZ = 0;
+        if (jumpTimer <= 0) {
+            dxr.Player.JumpZ = dxr.Player.Default.JumpZ;
+            PlayerMessage("Your knees feel fine again.");
+        }
+    }
+
+    if (speedTimer>0) {
+        speedTimer-=1;
+        dxr.Player.Default.GroundSpeed = DefaultGroundSpeed * moveSpeedModifier;
+        if (speedTimer <= 0) {
+            dxr.Player.Default.GroundSpeed = DefaultGroundSpeed;
+            PlayerMessage("Back to normal speed!");
+        }
+    }
+
+    if (lamthrowerTimer>0) {
+        lamthrowerTimer-=1;
+        if (lamthrowerTimer <=0) {
+            UndoLamThrowers();
+            PlayerMessage("Your flamethrower is boring again");
+        }
+    }
+    
+    if (iceTimer>0) {
+        iceTimer-=1;
+        if (iceTimer<=0) {
+            SetIcePhysics(False);
+            PlayerMessage("The ground thaws");
+        }
+    }
+    
+    if (behindTimer>0) {
+        behindTimer -=1;
+        if (behindTimer<=0){
+            dxr.Player.bBehindView=False;
+            PlayerMessage("You re-enter your body");
+        }
+    }
+    
+    if (difficultyTimer>0) {
+        difficultyTimer -=1;
+        if (difficultyTimer<=0){
+            dxr.Player.MPDamageMult=1.0;
+            PlayerMessage("Your body returns to its normal toughness");
+        }
+    }
+}
+
+function bool isCrowdControl( string msg) {
+    local string tmp;
+    //Validate if it looks json-like
+    if (InStr(msg,"{")!=0){
+        //PlayerMessage("Message doesn't start with curly");
```

Comment:
You could use the `info` function from the bottom of this file, or preferably a new function that does the same thing but called `warning`

Reply:
Yeah, I was just thinking this morning (before reading this comment even!) that it would probably be a good idea to send a telemetry message at least if we get a malformed packet, just in case Crowd Control changes the format of the message (or adds new things to it).